crates/y-sweet-core/src/auth.rs

```diff
@@ -1,18 +1,56 @@
-use std::fmt::Display;
-
-use base64::{
-    alphabet::{STANDARD, URL_SAFE},
-    engine::{DecodePaddingMode, GeneralPurpose, GeneralPurposeConfig},
-    Engine,
-};
 use bincode::Options;
+use data_encoding::Encoding;
 use rand::Rng;
 use serde::{Deserialize, Serialize};
 use sha2::{Digest, Sha256};
+use std::fmt::Display;
 use thiserror::Error;
 
 const EXPIRATION_MILLIS: u64 = 1000 * 60 * 60; // 60 minutes
 
+/// Static version of:
+///
+/// ```
+/// use data_encoding::{Specification, BASE64URL_NOPAD, Translate};
+/// let spec = Specification {
+///     ignore: "=".to_string(),
+///     translate: Translate {
+///         from: "/+".to_string(),
+///         to: "_-".to_string(),
+///     },
+///     ..BASE64URL_NOPAD.specification()
+/// };
+/// use y_sweet_core::auth::BASE64_CUSTOM;
+/// assert_eq!(BASE64_CUSTOM, spec.encoding().unwrap());
+/// ```
+pub const BASE64_CUSTOM: Encoding = Encoding::internal_new(&[
```

Comment:
What's the reason for inlining this, rather than having this in our code?

```rs
let spec = Specification {
    ignore: "=".to_string(),
    translate: Translate {
        from: "/+".to_string(),
        to: "_-".to_string(),
    },
    ..BASE64URL_NOPAD.specification()
};
// ...
spec.encoding().unwrap()
```

Reply:
Just performance. This would run every time we encoded/decoded a string (at least once per request) and involves at least three heap allocations (one for each to_string).

It’s arguably a premature optimization, but it is a low-cost one that I’m happy to make now.